### PR TITLE
Update subscriptions

### PR DESCRIPTION
**Pull request by prachigarg19**
1. When the user selects a date for the cancellation, the Frontend Client needs to calculate how many times we will still deduct the money via the selected payment method from their account. This way we want to ensure that the donor has all necessary information right away and can keep track.
2. Omit the option to edit the date, just leave Amount and Frequency

Note: For reference, the old PR is #1706

### CODE REVIEW

File: public/static/locales/en/me.json [hunk 1]
```diff
@@ -249,6 +249,9 @@
   "apiKeyMessage3": "Regenerating an API Key will invalidate your existing API Key immediately.",
   "regenerateKey": "Regenerate Key",
   "giftFund": "GiftFund",
+  "donatedTrees": "Donated Trees",
+  "remainingPaymentsText": "{{remainingPayments}} donations would take place before cancellation.",
+  "continue": "Continue"
```
Comment (sagararyal) [inline public/static/locales/en/me.json:254]: @nirali25parekh comma is missing here.

Comment (prachigarg19) [reply public/static/locales/en/me.json]: @sagararyal The deployment failed even after fixing the error.

**Comment by mohitb35** [comment]
Adjust the position of the popup shown before confirming cancellation. It should be in the centre of the screen.

Screenshot 2023-04-28 at 11 05 19 AM
Screenshot 2023-04-28 at 11 05 15 AM

**Comment by mohitb35** [comment]
This typo (seen while reactivating a subscription) should be corrected. It should say "billed" or "next billed", not "built". The date format may also need to be adjusted.

Screenshot 2023-04-28 at 11 05 59 AM

**Comment by mohitb35** [comment]
After cancelling a subscription on a future date, this is the UI seen. While this is technically correct information, we should also continue to show the next payment date (if it exists). 

Screenshot 2023-04-28 at 11 03 34 AM

File: src/features/user/Account/EditModal.tsx
Comment (mohitb35) [inline src/features/user/Account/EditModal.tsx:1]: @prachigarg19 Please clean up the commented code that isn't needed any more.

File: src/features/common/Layout/CustomModal/index.tsx [hunk 2]
```diff
@@ -6,10 +6,11 @@ import styles from './CustomModal.module.scss';
 interface Props {
   isOpen: boolean;
   handleContinue: () => void;
-  handleCancel: () => void;
+  handleCancel?: () => void;
   buttonTitle: string;
   modalTitle: string;
   modalSubtitle: string;
+  isCancel: boolean;
```
Comment (mohitb35) [inline src/features/common/Layout/CustomModal/index.tsx:13]: Is `isCancel` actually needed?

You can check whether `handleCancel` is undefined instead.

If you choose to keep this, rename `isCancel` to a more relevant name e.g. `isCancellable`, or `shouldAllowCancel` ?

File: src/features/user/Account/CancelModal.tsx
Comment (mohitb35) [inline src/features/user/Account/CancelModal.tsx:1]: While cancelling the donation on a future date, I don't see any loading/processing state on the button like for the other actions (e.g. reactivating the subscription).

File: src/features/user/Account/Recurrency.tsx [hunk 3]
```diff
@@ -134,6 +134,7 @@ export default function Recurrency({
                 handleCancelModalClose={handleCancelModalClose}
                 record={currentRecord}
                 fetchRecurrentDonations={fetchRecurrentDonations}
+                handleCancelModalOpen={() => setcancelModalOpen(true)}
```
Comment (mohitb35) [inline src/features/user/Account/Recurrency.tsx:137]: I'm not clear on why this is needed. Let's discuss this.

**Review by mohitb35 — CHANGES_REQUESTED** [review]

**Comment by mariahosfeld** [comment]
Will be implemented differently.

**Pull request closed without merging**